src/main/java/io/zold/api/RtTransaction.java

```diff
@@ -62,13 +67,29 @@
         this.transaction = trnsct;
     }
 
-    // @todo #15:30min Implement id() by parsing the string representation
-    //  of transaction according to the pattern, described in the white
-    //  paper. Replace relevant test case with actual tests.
     @Override
     @SuppressWarnings("PMD.ShortMethodName")
-    public long id() {
-        throw new UnsupportedOperationException("id() not yet implemented");
+    public long id() throws IOException {
+        try {
+            final String ident = new ItemAt<>(
+                new SplitText(this.transaction, ";").iterator(),
+                //@checkstyle MagicNumber (12 lines)
+                0
+            ).value().asString();
+            if (!RtTransaction.IDENT.matcher(ident).matches()) {
+                throw new IllegalArgumentException(
```

Comment:
@carlosmiranda why should it be an `IllegalArgumentException`? I think it is ok to simply use the `IOException` here instead of having this one wrapped inside another exception.